server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -150,4 +201,68 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
 
     }
+
+    private void validate(
+        ClusterState currentState,
+        Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap,
+        boolean restoreAllShards
+    ) {
+        String errorMsg = "cannot restore index [%s] because an open index with same name already exists in the cluster.";
+        for (Map.Entry<String, Tuple<Boolean, IndexMetadata>> indexMetadataEntry : indexMetadataMap.entrySet()) {
+            String indexName = indexMetadataEntry.getKey();
+            IndexMetadata indexMetadata = indexMetadataEntry.getValue().v2();
+            String indexUuid = indexMetadata.getIndexUUID();
+            boolean fromRemoteStore = indexMetadataEntry.getValue().v1();
+            if (indexMetadata.getSettings().getAsBoolean(SETTING_REMOTE_STORE_ENABLED, false)) {
+                if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState()) == false) {
+                    throw new IllegalStateException(String.format(Locale.ROOT, errorMsg, indexName) + " Close the existing index.");
+                }
+
+                if (fromRemoteStore) {
+                    boolean sameNameIndexExists = currentState.metadata().hasIndex(indexName);
+                    boolean sameUUIDIndexExists = currentState.metadata()
+                        .indices()
+                        .values()
+                        .stream()
+                        .anyMatch(indMd -> indMd.isSameUUID(indexUuid));
+                    if (sameNameIndexExists || sameUUIDIndexExists) {
+                        throw new IllegalStateException(String.format(Locale.ROOT, errorMsg, indexName));
+                    }
+                    Version minIndexCompatibilityVersion = currentState.getNodes().getMaxNodeVersion().minimumIndexCompatibilityVersion();
+                    metadataIndexUpgradeService.upgradeIndexMetadata(indexMetadata, minIndexCompatibilityVersion);
+                    boolean isHidden = IndexMetadata.INDEX_HIDDEN_SETTING.get(indexMetadata.getSettings());
+                    createIndexService.validateIndexName(indexName, currentState);
+                    createIndexService.validateDotIndex(indexName, isHidden);
+                    createIndexService.validateIndexSettings(indexName, indexMetadata.getSettings(), false);
+                }
+            } else {
+                logger.warn("Remote store is not enabled for index: {}", indexName);
+            }
+        }
+    }
+
+    private IndexMetadata getRemoteIndexMetadata() {
```

Comment:
this would be removed once we have the download flow in place. Draft PR for download flow https://github.com/soosinha/OpenSearch/pull/2/files